rotkehlchen/chain/evm/decoding/hop/balances.py

```diff
@@ -0,0 +1,165 @@
+from rotkehlchen.history.events.structures.evm_event import EvmProduct
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_evm_address
+from rotkehlchen.types import ChecksumEvmAddress, EvmTokenKind
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class HopCommonBalances(ProtocolWithBalance):
+    def __init__(
+            self,
+            database: 'DBHandler',
+            evm_inquirer: 'EvmNodeInquirer',
+    ):
+        super().__init__(
+            database=database,
+            evm_inquirer=evm_inquirer,
+            counterparty=CPT_HOP,
+            deposit_event_types={(HistoryEventType.STAKING, HistoryEventSubType.DEPOSIT_ASSET)},
+        )
+
+    def query_balances(self) -> 'BalancesSheetType':
+        """Queries and returns the balance sheet for staking events.
+
+        This method:
+        1. Retrieves deposit events for staking.
+        2. Groups addresses by staking contract.
+        3. For each contract, queries user balances and rewards.
+        4. Converts balances and rewards to USD and updates the balance sheet."""
+        balances: BalancesSheetType = defaultdict(BalanceSheet)
+        if len(addresses_with_deposits := self.addresses_with_deposits(products=[EvmProduct.STAKING])) == 0:  # noqa: E501
+            return balances
+
+        # Group addresses by staking contract
+        staked_contracts: dict[ChecksumEvmAddress, set[ChecksumEvmAddress]] = defaultdict(set)
+        for user_address, events in addresses_with_deposits.items():
+            for event in events:
+                if event.address is not None:
+                    staked_contracts[event.address].add(user_address)
+
+        hop_abi = self.evm_inquirer.contracts.abi('HOP_STAKING')
+        for contract_address in staked_contracts:
+            addresses = staked_contracts[contract_address]
+            staking_contract = EvmContract(
+                address=contract_address,
+                abi=hop_abi,
+                deployed_block=0,
+            )
+            chunk_size, call_order = get_chunk_size_call_order(self.evm_inquirer)
+
+            if len(staked_lps := self.evm_inquirer.multicall(
+                calls=[(
+                    staking_contract.address,
+                    staking_contract.encode(
+                        method_name='balanceOf',
+                        arguments=[user],
+                    ),
+                ) for user in addresses],
+                call_order=call_order,
+                calls_chunk_size=chunk_size,
+            )) == 0:
+                log.warning('No staked balances found for Hop. Skipping')
```

Comment:
When I say log a warning it needs to have more info

```suggestion
                log.warning('No staked balances found for {self.evm_inquirer.WHATEVER_VAR_IS_CHAIN_NAME} Hop contract {staking_contract.address}. Skipping')
```